File: Fines2Model3/Data_/Asignatura.cs
using SqlOrganize.CollectionUtils;
using SqlOrganize.Sql;

namespace SqlOrganize.Sql.Fines2Model3
{
    public partial class Asignatura
    {

        public override string? Label
        {
            get
            {
                if (!_Label.IsNoE())
                    return _Label;

                return nombre ?? "?" + " " + codigo ?? "?";
            }
            set
            {
                Label = value;
                NotifyPropertyChanged(nameof(Label));
            }
        }

    }
}

File: Model/Asignatura_.php
<?php

namespace Fines2;

use \Fines2\Asignatura;

use SqlOrganize\Sql\Entity;
use Exception;
use DateTime;

class Asignatura_ extends Asignatura
{
    public function getLabel(): string{
        return mb_convert_encoding($this->nombre . " " . $this->codigo ?? "?", 'UTF-8', 'auto');
    }

}

